### PR TITLE
update: Update the ingress api

### PR DESCRIPTION
**Pull request by harrisonliu5**
Signed-off-by: harrisonliu5 <harrisonliu_5@163.com>

### What type of PR is this?

/kind api-change

### Which issue(s) this PR fixes:

Fixes ##https://github.com/kubesphere/kubesphere/issues/4068

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
Update the ingress api
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/sample/bookinfo.yaml [hunk 1]
```diff
@@ -348,7 +348,7 @@ spec:
       targetPort: 9080
 
 ---
-apiVersion: extensions/v1beta1
+apiVersion: "networking.k8s.io/v1"
```
Comment (RolandMa1986) [inline server/sample/bookinfo.yaml:351]: No only the `apiVersion` need to be changed. but also the rules structure. 
```suggestion
apiVersion: "networking.k8s.io/v1"
kind: Ingress
metadata:
  namespace: servicemesh
  labels:
    app.kubernetes.io/version: v1
    app.kubernetes.io/name: bookinfo
  name: bookinfo-ingress
spec:
  rules:
    - http:
        paths:
          - path: /
            backend:
              service:
                name: productpage
                port: 9080
      host: productpage.servicemesh.139.198.121.92.nip.io
```

Comment (harrisonliu5) [reply server/sample/bookinfo.yaml:351]: update

**Comment by zheng1** [comment]
@harrisonliu5 Please fix the conflict

**Pull request merged**